graph/push.go

```diff
@@ -515,7 +515,7 @@ func (s *TagStore) CmdPush(job *engine.Job) engine.Status {
 		return job.Errorf("Repository does not exist: %s", repoInfo.LocalName)
 	}
 
-	if endpoint.Version == registry.APIVersion2 {
+	if repoInfo.Index.Official || endpoint.Version == registry.APIVersion2 {
```

Comment:
Currently there is too much special casing going on in the code, so the official registry is not using the same method to detect v1/v2.  The fix for this is a deep refactor which will be part of the code migration to the distribution package. This case is making the index a v2 registry despite the endpoint version, there is additional code later on which then explicitly uses the v2 registry endpoint when making v2 requests.